src/tests/system/tests/test_default_debug_level.py

```diff
@@ -0,0 +1,191 @@
+    assert start_log_length == log_length(), "Debug messages were generated"
+
+
+@pytest.mark.ticket(bz=1893159)
+@pytest.mark.topology(KnownTopology.Client)
+def test_default_debug_level__fatal_and_critical_failures(client: Client):
+    """
+    :title: Check that messages with levels 0 and 1 are logged for fatal or critical failures
+    :setup:
+        1. Start SSSD with default debug level (config file is created)
+        2. Restrict sssd.conf permissions
+    :steps:
+        1. Restart sssd
+        2. Check logs
+    :expectedresults:
+        1. SSSD failed to start
+        2. Fatal failures (level 0) and Critical failures (level 1) are in log file
+    :customerscenario: True
+    """
+    client.sssd.common.local()
+    client.sssd.default_domain = "local"
+    client.sssd.start(debug_level=None)
+    client.fs.chmod(mode="444", path="/etc/sssd/sssd.conf")
+
+    assert (
+        client.sssd.restart(debug_level=None, raise_on_error=False, apply_config=False).rc != 0
+    ), "SSSD started successfully, which is not expected"
+
+    log_str = client.fs.read("/var/log/sssd/sssd.log")
+    assert "0x0010" in log_str, "Fatal failures were not found in log"
+    assert "0x0020" in log_str, "Critical failures were not found in log"
+
+
+@pytest.mark.ticket(bz=1893159)
+@pytest.mark.topology(KnownTopology.Client)
+def test_default_debug_level__cannot_load_sssd_config(client: Client):
+    """
+    :title: Check that messages with level 2 are logged when SSSD can't load config
+    :setup:
+        1. Set 'domains' to 'non_existing_domain' in sssd section
+    :steps:
+        1. Try to start SSSD with default debug level
+        2. Check logs
+    :expectedresults:
+        1. SSSD failed to start
+        2. Correct error message is in log file
+    :customerscenario: True
+    """
+    client.sssd.sssd["domains"] = "non_existing_domain"
+    assert (
+        client.sssd.start(debug_level=None, raise_on_error=False).rc != 0
+    ), "SSSD started successfully, which is not expected"
+    assert "SSSD couldn't load the configuration database" in client.fs.read("/var/log/sssd/sssd.log")
+
+
+@pytest.mark.ticket(bz=1893159)
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_default_debug_level__nonexisting_ldap_server(client: Client):
+    """
+    :title: Check that messages with level 2 are logged when LDAP server doesn't exist
+    :setup:
+        1. Start sssd with default debug level
+        2. Set ldap_uri to a non-existing ldap-server
+    :steps:
+        1. Check logs
+        2. Check default domain status
+    :expectedresults:
+        1. Domain logs should contain a log related to 'going offline'
+        2. LDAP is not connected
+    :customerscenario: True
+    """
+    client.sssd.domain["ldap_uri"] = "ldap://typo.invalid"
+    client.sssd.start(debug_level=None, raise_on_error=False)
+
+    time.sleep(10)
+    logs = client.fs.read(f"/var/log/sssd/sssd_{client.sssd.default_domain}.log")
```

Comment:
Use https://tests.sssd.io/en/latest/api/sssd_test_framework.utils.sssd.html#sssd_test_framework.utils.sssd.SSSDUtils.logs

This applies to all places where you read the logs.

Reply:
Okay, and what about ``"/var/log/sssd/sssd.log`` files. We do not have it in framework, should I add it?

Reply:
Yes, please. As "monitor".

Reply:
PR created: https://github.com/SSSD/sssd-test-framework/pull/57